pkg/github/internal/cache/cache.go

```diff
@@ -0,0 +1,84 @@
+package cache
+
+import (
+	"fmt"
+	"time"
+
+	gocache "github.com/patrickmn/go-cache"
+
+	"github.com/giantswarm/config-controller/pkg/github/internal/gitrepo"
+)
+
+const (
+	defaultExpiration = 0
+)
+
+type StoreCache struct {
+	underlying *gocache.Cache
+}
+
+func NewStoreCache(expiration time.Duration) *StoreCache {
+	c := &StoreCache{
+		underlying: gocache.New(expiration, expiration/2),
+	}
+	return c
+}
+
+func (c *StoreCache) Set(owner, name, ref string, value *gitrepo.Store) {
```

Comment:
WDYT about creating `Key` struct with `Owner`, `Name` and `Ref` fields? It feels that the list of string params is quite long here and repeated in many places.

Reply:
I've experimented with that, but this combination or parts of it are set in 2-3 places tops. Plus it would seem weird to use:
- only some parts of the structure + other fields (e.g. don't use reference, use major)
- have functions which use the structure, and functions which define `owner` and `name` separately